packages/react-server/src/ReactFizzServer.js

```diff
@@ -222,6 +224,7 @@ type RenderTask = {
   childIndex: number,
   ping: () => void,
   blockedBoundary: Root | SuspenseBoundary,
+  blockedBoundaryState: null | BoundaryState, // Boundary state we'll mutate while rendering. This may not equal the state of the blockedBoundary
```

Comment:
This is tricky. The blocked boundary is the one that can't actually be completed until all the tasks are done. For a fallback task this is the parent of the boundary the fallback is for. But for `blockedBoundaryState` it could the the state of the blockedBoundary (usually the case) but it could also be the `fallbackState` of the deeper boundary when the task is for generating the fallback. It's not really blocked so the terminology isn't quite right.

I think maybe renaming 
* `BoundaryState` to `HoistableState` and
* `blockedBounaryState` to `hoistableState`
Which may help with understanding